general/device.cpp

```diff
@@ -150,6 +151,19 @@ static void CudaDeviceSetup(const int dev, int &ngpu)
 #endif
 }
 
+static void HipDeviceSetup(const int dev, int &ngpu)
+{
+#ifdef MFEM_USE_HIP
+   int deviceId;
+   MFEM_GPU_CHECK(hipGetDevice(&deviceId));
+   hipDeviceProp_t props;
+   MFEM_GPU_CHECK(hipGetDeviceProperties(&props, deviceId));
+   printf("info: running on device #%d %s\n", deviceId, props.name);
```

Comment:
Use 'mfem::out' or 'mfem::err' instead of 'printf'.

Reply:
done